sympy/functions/special/delta_functions.py

```diff
@@ -159,6 +159,31 @@ def is_simple(self, x):
             return p.degree() == 1
         return False
 
+    def _eval_rewrite_as_Piecewise(self, arg):
+        """Represents DiracDelta in a Piecewise form
+
+           Examples
+           ========
+
+           >>> from sympy import DiracDelta, Piecewise, Symbol, oo
+           >>> x = Symbol('x', real=True)
+
+           >>> DiracDelta(x).rewrite(Piecewise)
+           Piecewise((oo, Eq(x, 0)), (0, True))
+
+           >>> DiracDelta(x - 5).rewrite(Piecewise)
+           Piecewise((oo, Eq(x - 5, 0)), (0, True))
+
+           >>> y = Symbol('y')
+
+           >>> DiracDelta(y).rewrite(Piecewise)
+           DiracDelta(y)
+
+        """
+        from sympy import oo
```

Comment:
Since there is already one `from sympy import ...`. Also it is generally a good idea to make import of singletons at the top of the file. Functions like `sin`, `cos`, `tan`, `S`, `diff` are also good to be used a top-level imports since, they are often used.
The code-level import should generally only be made when there is problem of **cyclic import** arises. This no hard and fast rule, but is good to follow, since it keeps multiple imports of a function at code-level out of the picture.